lang/py/src/avro/io.py

```diff
@@ -475,8 +591,22 @@ def read_data(self, writers_schema, readers_schema, decoder):
     elif writers_schema.type == 'double':
       return decoder.read_double()
     elif writers_schema.type == 'bytes':
-      return decoder.read_bytes()
+      if (hasattr(writers_schema, 'logical_type') and
+                      writers_schema.logical_type == 'decimal'):
```

Comment:
Please avoid `hasattr` and use `getattr` instead. Hynek Schlawak has a [nice blog post](https://hynek.me/articles/hasattr/) about why.